### ISSUE TITLE
Input issues with DDNet 10.3.2

### ISSUE THREAD

**Issue by Xandaros** [ddnet/ddnet#572]
Since updating to 10.3.2, I have some really bizarre behaviour with my input.
This has not happened with older versions and is also not present when using the vanilla client.

When I open chat, it comes with a t pre-typed. (I open chat with t) If I enter a t into chat, however, only one appears.
Deleting characters in chat with backspace removes 2 at a time. Similarly, moving the cursor using the arrow keys, moves two characters at a time. Deleting with the delete key also removes 2 at once.
The ESC key, as well as the F-keys don't actually do anything unless I press right shift beforehand.
The significance of right shift is a bit lost to me, but it might be because right shift is my compose key.

I am using Linux with i3 as my window manager. For the compose key I actually use ibus,  which I am starting in my .xinitrc like this:

```
export GTK_IM_MODULE=ibus
export XMODIFIERS=@im=ibus
export QT_IM_MODULE=ibus

ibus-daemon -x -d -r
```

I am not sure if this is significant, but I thought I'd mention it. ~~I have not yet tried it without ibus running, but I will shortly. If you see this issue right after I submitted it, you can expect it to be edited within 10 minutes.~~
Yes, definitely something with ibus. If I comment out the lines above, I don't have any issues.

Edit1:
I bisected a bit and it seems the first commit in which it happens is one of these:
5a42d90ee680dc8c5e27b0998b6752369333cae2
3e9b8a62cec7c939d22efdd15485e6de1f1f057d
41f0365de1bab4085ea261217d1c54660a078829
632db36fe407ff09352954a19082d433e612fd78
ea451736125957fe2420304aa0225321a6ddd368
0e609cd92c002d293662b6b9e95b38e6f3b58684
9918ada2d865e6f7fdaefbd59eb43c9439861e19
4111acd3098b0024a0b189a85b740c9d4223c98a
c5db3ef5f41648874b3c85c31c57fc9d4094d598
8bfc0008b78905260530f70dfed5ef0296f96698
bd2b61f0e5e9e83cb33dc5dd3b87ba3e49be1ac7
6efd028393c36c5d01ef3fc1916db75e1c6a9f19
a1403ef4579f4b555b8a8783233c15d9ce63ca86
38cfe323858a529d28ba6e240aebb3b2efaf9c5f
a089d3d36d2e290c8e010c327749cd1ee1472d59
5cdb36ecb02bda5952e3e034ad5857be4eff2a8d
cc1f80471bfc093b745b30927799711521416efe
0155c75cf44e3d3a524aee3e0b3fefd2098ab2e8
ff2a7272fa4a49dec7a1e57a30ef32742667c435
10d6f69d32586544e607f2057fc25418c881565e
63a4c804de170f74bad6459d6caa3073eab88fd9
8e3550f9c675238e8154a32727701faa3894f6b3
ebb595363280ae60cb2a5259c44c8c7391372bf8
2d70c3e292c495a3ed5894684b09a380bd8cfd57
6c4e6c5c44715f39f7471c6c2eab388d23f5674d
0554b590e8a29a537349c3c9f90dc2a1fd5c2fd2
354696afb7b3fbb2c9f9f706377238c0d1b24553
db8a126315de1657253917b861b51306da9e3a53
e8965bee68711b08a309d51801b99363f0daf733
2e0ca0f4b435ec07e3c92264c9a322026be359d

I can't actually narrow it down further, because none of those actually compile... How does one manage to go so many commits without being able to compile? o_O

Anyway, those are all between 9.3 and 10. I think this is somehow related to the switch to SDL2, which would explain why the vanilla client and older versions are unaffected.
However, I have been using newer versions just fine until now.

The last time ibus was updated on my system was on the 14th of August, so I don't think this caused it. It has been a while since I last played, but I don't think it was that long. (I could be wrong, though)
Now, sdl2 on the other hand was last updated on the 23rd of October. This is much closer to the current date and might have triggered this. The question now becomes: Is it sdl2's fault or is it DDNet's fault?
I will investigate a bit further, but I need to go to bed soon^^

Edit2:
It appears this only happens with SDL 2.0.5. Downgrading to SDL 2.0.4 does indeed fix this issue.
For the time being, I shall stay on 2.0.4. I think this is actually an SDL issue, but I will keep the issue open for now. (And hope that you people complain to the SDL folks :P)

**Comment by awilliamson** [comment]
Can confirm that "sdl2-2.0.4-2" ( On Arch Linux ) works with DDNet Client.

**Comment by PathosEthosLogos** [comment]
I think this can be marked as resolved, correct?

**Comment by Xandaros** [comment]
Nope. I just tried updating sdl again and it still happens.

This might be caused by https://bugzilla.libsdl.org/show_bug.cgi?id=3155

I do believe this is an SDL bug, but it would be a bit odd for me, as a user, to report it. Especially since I probably wouldn't be able to answer any follow-up questions.

**Comment by def-** [comment]
I tried it out with ibus and recent SDL2 version and can't reproduce. If you can reproduce this with the most recent SDL2 version, then reopen and explain how to reproduce it in more detail.

**Comment by Xandaros** [comment]
I just tried it again after updating and, indeed, this is now fixed.
If you didn't do anything, I guess it was fixed in sdl2? Still on 2.0.5, though. The actual package was changed, however. Maybe something with how it was packaged.

In any case, it's fixed. Just wanted to say that :)

**Comment by def-** [comment]
You could check version 10.6.4. I have some fix for text input in 10.6.5: https://github.com/ddnet/ddnet/commit/f627a72ebb7ad325b1fbba49d4988a42d0ca5c6e
